### ISSUE TITLE
Decide on an upper bound for data commitment windows

### ISSUE THREAD

**Issue by SweeXordious** [celestiaorg/celestia-core#1097]
Currently, the upper bound for data commitment window is 1000 in [here](https://github.com/celestiaorg/celestia-core/blob/13b9e92ba0bb545ab853380e34e30150016b9413/pkg/consts/consts.go#L43C1-L44):

```golang
// DataCommitmentBlocksLimit is the limit to the number of blocks we can generate a data commitment for.
DataCommitmentBlocksLimit = 1000
```

After discussing https://github.com/celestiaorg/quantum-gravity-bridge/issues/181#issuecomment-1736078493, we should decide whether we want to increase that value higher since changing it after launch will require a hardfork.

cc @evan-forbes @adlerjohn

**Comment by evan-forbes** [comment]
the benchmark is cool and its good to know that it doesn't take an insane amount of time for empty blocks. 

> Note: the blocks are empty in this benchmark.

However wouldn't full blocks give us a way better understanding of if this is an issue or not?

What about the estimated cost in ETH for the relayer graph that you showed me earlier @SweeXordious ? that seems like a bigger factor when picking data commitment windows, no?

**Comment by SweeXordious** [comment]
> However wouldn't full blocks give us a way better understanding of if this is an issue or not?

Yes, but it's hard to benchmark with full blocks... we need a realistic network and all which is not easy to get in a benchmark environment. We could do it via inspecting the resources usage but it's not gonna be easy to do.

Any ideas? 

> What about the estimated cost in ETH for the relayer graph that you showed me earlier @SweeXordious ? that seems like a bigger factor when picking data commitment windows, no?

Yes, I will include it here, thanks
